dev-packages/application-package/src/extension-package-collector.ts

```diff
@@ -73,7 +73,7 @@ export class ExtensionPackageCollector {
         if (RawExtensionPackage.is(pck)) {
             const parent = this.parent;
             const version = pck.version;
-            const transitive = !(name in this.root.dependencies!);
+            const transitive = !(name in this.root!.dependencies!);
```

Comment:
My guess is that if we had the strict checks enabled before, this method wouldn't have been implemented without this oddity in the first place. So not sure what to think about this code indeed. What I like right now is that it properly describes how `root` can be `undefined` but that this place expects it not to be.

Reply:
In the rare case that this could become undefined in the future, I believe that having some sort of guard for that would be great. Something as simple as `if (!this.root) { return; }`